src/main/java/Calculator.java

```diff
@@ -1,86 +1,233 @@
+import java.util.*;
+import java.util.logging.Logger;
+
+import static util.CalculatorUtil.*;
+
 public class Calculator {
 
-    public int add(int a, int b) {
-        validateOperation(a, b, Operator.ADD);
+    private static final Set<String> BASIC_SEPARATORS = CalculatorConstant.BASIC_SEPARATORS;
+    private static final String CUSTOM_SEPARATOR_STARTER = CalculatorConstant.CUSTOM_SEPARATOR_STARTER;
+    private static final String CUSTOM_SEPARATOR_TERMINATOR = CalculatorConstant.CUSTOM_SEPARATOR_TERMINATOR;
+    private static final Set<Character> REGEX_META_CHARACTERS = CalculatorConstant.REGEX_META_CHARACTERS;
+    private static final String BASIC_REGEX = createBasicRegex(BASIC_SEPARATORS);
+
+    private final Logger logger = Logger.getLogger("Calculator Logger");
+
+    public int add(int leftOperand, int rightOperand) {
+        return Operator.ADD.operate(leftOperand, rightOperand);
+    }
 
-        return a + b;
+    public int subtract(int leftOperand, int rightOperand) {
+        return Operator.SUBTRACT.operate(leftOperand, rightOperand);
     }
 
-    public int subtract(int a, int b) {
-        validateOperation(a, b, Operator.SUBTRACT);
+    public int multiply(int leftOperand, int rightOperand) {
+        return Operator.MULTIPLY.operate(leftOperand, rightOperand);
+    }
 
-        return a - b;
+    public int divide(int dividend, int divisor) {
+        return Operator.DIVIDE.operate(dividend, divisor);
     }
 
-    public int multiply(int a, int b) {
-        validateOperation(a, b, Operator.MULTIPLY);
+    /**
+     * 구분자를 가진 문자열을 받아 모든 숫자의 합을 반환한다. <br>
+     * 쉼표 또는 콜론을 기본 구분자로 사용한다. <br>
+     * 커스텀 구분자를 사용하고 싶을 경우, 문자열 앞부분에 "//${커스텀구분자}\n"을 추가한다. <br>
+     * ex_ "//;\n1;2;3" <br>
+     *
+     * @param input 구분자로 구분된 자연수 양수를 지닌 문자열
+     * @return 모든 숫자의 합
+     * @throws RuntimeException 구분자가 부적절함<br>숫자 이외의 값이 전달됨<br>음수가 전달됨
+     */
+    public int addAll(String input) {
+        List<Integer> numbers;
+        logger.info("입력된 값: " + input);
+
+        if (isStartsWithCustomSeparatorStarter(input)) {
+            String customSeparator = getCustomSeparatorFromInput(input);
+            String extractedInput = removeCustomSeparatorDeclarationFromInput(input);
+            String safetyCustomSeparator = getSafetyCustomSeparator(customSeparator);
 
-        return a * b;
+            String regex = createRegexByCustomSeparator(safetyCustomSeparator);
+            logger.info("생성된 Regex: " + regex);
+            numbers = convertStringNumbersIntoList(extractedInput, regex);
+        } else {
+            logger.info("BASIC_REGEX 사용: " + BASIC_REGEX);
+            numbers = convertStringNumbersIntoList(input, BASIC_REGEX);
+        }
+
+        int sum = 0;
+        for (Integer number : numbers) {
+            sum += number;
+        }
```

Comment:
### 생각해보면 좋을 부분

`sum += number;` 구문에서 `number`는 `Integer` 타입이고 `sum`은 `int` 타입입니다. 이때 어떻게 연산이 될까요? 반복문을 돌면서 매번 `언박싱`이 발생한다면, 불필요한 연산이 추가되어 성능 저하가 발생할 수 있는 코드일까요?

Reply:
```suggestion
        Integer sum = 0;
        for (Integer number : numbers) {
            sum += number;
        }
```
컬렉션에 담긴 primitive type은 자동으로 Wrapper 클래스로 변환되므로, int 타입인 sum에 더해질때마다 언박싱이 발생해 성능 저하가 일어나겠군요...!
위와 같이 수정하는 것이 바람직해 보이네요.
피드백 감사합니다!